test/cmd/newapp.sh

```diff
@@ -9,10 +9,18 @@ source "${OS_ROOT}/hack/util.sh"
 source "${OS_ROOT}/hack/cmd_util.sh"
 os::log::install_errexit
 
+# Cleanup cluster resources created by this test
+(
+  set +e
+  oc delete all,templates --all
+  exit 0
+) &>/dev/null
```

Comment:
What I'm doing is starting up the master in another cmd shell (openshift
start master --loglevel=5) and then just running test/cmd/newapp.sh.  A few
tests need preconditions, but any test that needs preconditions should set
them up anyway (for test-cmd).

On Tue, Feb 2, 2016 at 11:00 AM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In test/cmd/newapp.sh
> <https://github.com/openshift/origin/pull/6670#discussion_r51588259>:
>
> > @@ -9,10 +9,18 @@ source "${OS_ROOT}/hack/util.sh"
> >  source "${OS_ROOT}/hack/cmd_util.sh"
> >  os::log::install_errexit
> >
> > +# Cleanup cluster resources created by this test
> > +(
> > +  set +e
> > +  oc delete all,templates --all
> > +  exit 0
> > +) &>/dev/null
>
> The pattern I've seen before in these tests was to do clean ups only when
> necessary between test cases. The "test framework" automatically creates a
> project for each test/cmd/*.sh file and after the tests run it will
> remove the whole project.
>
> So the way I've been running these tests locally during development is:
>
> $ hack/test-cmd.sh test/cmd/newapp.sh
>
> Perhaps you're running them in a smarter way?
>
> What I don't like in the way I run it is that it takes a way too long time
> to start a server in every test run, making developing those tests quite
> painful. I often comment out all other tests in a file.
>
> @smarterclayton <https://github.com/smarterclayton> could you describe
> how you develop and run test/cmd tests locally? I'd happily add the
> knowledge to HACKING.md
> <https://github.com/openshift/origin/blob/master/HACKING.md#integration-tests>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6670/files#r51588259>.
>